app/plugin.go

```diff
@@ -886,6 +890,44 @@ func (a *App) processPrepackagedPlugin(pluginPath *pluginSignaturePath) (*plugin
 	return plugin, nil
 }
 
+// installFeatureFlagPlugins handles the automatic installation/upgrade of plugins from feature flags
+func (a *App) installFeatureFlagPlugins() {
+	ffControledPlugins := a.Config().FeatureFlags.Plugins()
+
+	// Respect the automatic prepackaged disable setting
+	if !*a.Config().PluginSettings.AutomaticPrepackagedPlugins {
```

Comment:
Since `AutomaticPrepackagedPlugins` defaults to `true`, does this mean we could technically trigger a remote installation of non-prepackaged plugins when an on-premise customer upgrades Mattermost? Is that desirable?

Reply:
Yes. I think that is the intended effect. Ideally we would remove / package separately the pre-packaged plugins and this would be the main mechanism for accomplishing what those do today. 
@itao can confirm.